src/circuits/commitments/commitments.hpp

```diff
@@ -43,59 +44,38 @@ libsnark::pb_variable_array<FieldT> getRightSideCMCOMM(
     libsnark::pb_variable<FieldT> &ZERO,
     libsnark::pb_variable_array<FieldT> &value_v);
 
-// TODO: Implement the COMM_k_gadget as a 2 hash rounds in order to directly get
-// the value of the commitment_k without needing 2 distinct gadgets for this.
-//
 // See Zerocash extended paper, page 22
-// The commitment k is computed as k = sha256(r || [sha256(a_pk || rho)]_128)
-// where we define the left part: inner_k = sha256(a_pk || rho)
-// as being the inner commitment of k
-template<typename FieldT, typename HashT>
-class COMM_inner_k_gadget : public COMM_gadget<FieldT, HashT>
-{
-public:
-    COMM_inner_k_gadget(
-        libsnark::protoboard<FieldT> &pb,
-        libsnark::pb_variable_array<FieldT>
-            &a_pk, // public address key, 256 bits
-        libsnark::pb_variable_array<FieldT> &rho, // 256 bits
-        std::shared_ptr<libsnark::digest_variable<FieldT>>
-            result, // sha256(a_pk || rho)
-        const std::string &annotation_prefix = "COMM_inner_k_gadget");
-};
-
-// See Zerocash extended paper, page 22
-// The commitment k is computed as k = sha256(r || [sha256(a_pk || rho)]_128)
-// where we define: outer_k = sha256(r || [inner_commitment]_128)
-// as being the outer commitment of k
+// The commitment cm is computed as
+// hashT(hashT( trap_r || [hashT(a_pk, rho)]_[128]) || "0"*192 || v)
 // We denote by trap_r the trapdoor r
 template<typename FieldT, typename HashT>
-class COMM_outer_k_gadget : public COMM_gadget<FieldT, HashT>
+class COMM_k_gadget : public libsnark::gadget<FieldT>
 {
-public:
-    COMM_outer_k_gadget(
-        libsnark::protoboard<FieldT> &pb,
-        libsnark::pb_variable_array<FieldT> &trap_r, // 384 bits
-        libsnark::pb_variable_array<FieldT>
-            &inner_k, // 256 bits, but we only keep 128 bits out of it
-        std::shared_ptr<libsnark::digest_variable<FieldT>>
-            result, // sha256(trap_r || [inner_k]_128)
-        const std::string &annotation_prefix = "COMM_outer_k_gadget");
-};
+private:
+    std::shared_ptr<libsnark::block_variable<FieldT>> block;
+    std::shared_ptr<COMM_gadget<FieldT, HashT>>
+        inner_com_gadget; // Hash gadget used as a commitment
```

Comment:
Please comment on the line above (rather then at the end of line) to avoid line breaks when we run clang-format

Reply:
done